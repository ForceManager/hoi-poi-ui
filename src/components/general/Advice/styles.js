export default (theme) => ({
    root: {
        ...theme.typography.defaultText,
        borderRadius: '8px',
        display: 'flex',
        padding: '16px',
        '& .ReactCollapse--collapse': {
            width: '100%',
            minHeight: 20,
            overflow: 'hidden !important',
            transition: 'height 500ms',
            willChange: 'height',
        },
    },
    icon: {
        display: 'flex',
        marginRight: 10,
        marginTop: 1,
    },
    Text: {
        display: 'block',
        width: '100%',
    },

    // Types
    default: {
        backgroundColor: theme.colors.neutral100,
        '& $Text': {
            color: theme.colors.neutral700,
        },
    },
    error: {
        backgroundColor: theme.colors.red100,
        '& $Text': {
            color: theme.colors.red500,
        },
    },
    info: {
        backgroundColor: theme.colors.aqua100,
        '& $Text': {
            color: theme.colors.aqua700,
        },
    },
    success: {
        backgroundColor: theme.colors.green100,
        '& $Text': {
            color: theme.colors.green700,
        },
    },
    warning: {
        backgroundColor: theme.colors.yellow100,
        '& $Text': {
            color: theme.colors.yellow700,
        },
    },
    textContainer: {
        flex: 1,
        overflow: 'hidden',
    },
    collapseContainer: {
        display: 'flex',
        overflow: 'hidden',
        alignItems: 'flex-start',
    },
    dropdownIcon: {
        lineHeight: '0',
        transition: 'transform 0.3s ease-out',
        willChange: 'transform',
        transform: 'rotate(-180deg) translateY(1px)',
        cursor: 'pointer',
    },
    isOpened: {
        '& $dropdownIcon': {
            transform: 'rotate(0) translateY(1px)',
        },
        '& $Text': {
            whiteSpace: 'pre-wrap',
        },
    },
    withTitleContainer: {
        marginTop: 15,
    },
});
