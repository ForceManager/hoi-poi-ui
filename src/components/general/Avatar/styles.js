export default (theme) => ({
    root: {
        ...theme.typography.defaultText,
        color: theme.colors.neutral700,
        overflow: 'hidden',
        background: theme.colors.neutral400,
        position: 'relative',
        '& > img': {
            height: '100%',
            width: '100%',
            objectFit: 'cover',
            textAlign: 'center',
        },
    },
    round: {
        borderRadius: '50%',
        '& $icon': {
            width: '80%',
            height: '80%',
            padding: '10%',
        },
    },
    square: {
        borderRadius: 4,
        '& $icon': {
            width: '96%',
            height: '96%',
            padding: '2%',
        },
    },
    tiny: {
        height: 18,
        width: 18,
        '& $initials': {
            fontSize: 10,
        },
        '& $count': {
            transform: 'translate3d(40%, 40%, 0)',
        },
    },
    small: {
        height: 20,
        width: 20,
        '& $initials': {
            fontSize: 10,
        },
        '& $count': {
            transform: 'translate3d(40%, 40%, 0)',
        },
    },
    medium: {
        height: 25,
        width: 25,
    },
    large: {
        height: 30,
        width: 30,
        '& $initials': {
            fontSize: 16,
        },
    },
    big: {
        height: 32,
        width: 32,
        '& $initials': {
            fontSize: 18,
        },
    },
    huge: {
        height: 64,
        width: 64,
        '& $initials': {
            fontSize: 38,
        },
        '& $count': {
            transform: 'none',
        },
    },
    icon: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    initials: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        width: '100%',
        height: '100%',
        textTransform: 'uppercase',
    },
    multiAvatar: {
        overflow: 'visible',
    },
    multiAvatarInner: {
        display: 'grid',
        overflow: 'hidden',
        borderRadius: '50%',
    },
    multiAvatarItem: {
        '& img': {
            width: '100%',
            height: '100%',
            objectFit: 'cover',
            display: 'block',
        },
        '&:first-child': {
            gridColumn: 1,
            gridRow: '1 / span 2',
            aspectRatio: 0.5,
        },

        '&:nth-child(2)': {
            gridColumn: 2,
            gridRow: 1,
            aspectRatio: 1,
        },

        '&:nth-child(3)': {
            gridColumn: 2,
            gridRow: 2,
            aspectRatio: 1,
        },

        '&:nth-child(2):is(:last-child)': {
            gridColumn: 2,
            gridRow: '1 / span 2',
            aspectRatio: 0.5,
        },

        '&:only-child': {
            gridColumn: '1 / span 2',
            gridRow: '1 / span 2',
            aspectRatio: 1,
        },
    },
    count: {
        position: 'absolute',
        bottom: 0,
        right: 0,
        boxSizing: 'border-box',
        boxShadow: `0 0 0 2px ${theme.colors.neutralBase}`,
        backgroundColor: theme.colors.neutral400,
        transform: 'translate3d(25%, 25%, 0)',
    },
});
