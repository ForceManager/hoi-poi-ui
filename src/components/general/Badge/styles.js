export default (theme) => ({
    root: {
        ...theme.typography.smallText,
        lineHeight: '20px',
        color: theme.colors.text.black,
        borderRadius: '3px',
        display: 'inline-block',
        padding: '0 10px',
        minWidth: 70,
        textAlign: 'center',
    },
    Text: {
        lineHeight: '20px',
        fontWeight: 500,
        textTransform: 'uppercase',
    },
    // Types
    error: {
        backgroundColor: theme.colors.redLight,
        '& $Text': {
            color: theme.colors.text.red,
        },
    },
    info: {
        backgroundColor: theme.colors.greyLight,
        '& $Text': {
            color: theme.colors.text.greySoft,
        },
    },
    success: {
        backgroundColor: theme.colors.greenLight,
        '& $Text': {
            color: theme.colors.text.green,
        },
    },
    warn: {
        backgroundColor: theme.colors.yellowLight,
        '& $Text': {
            color: theme.colors.text.yellow,
        },
    },
});
