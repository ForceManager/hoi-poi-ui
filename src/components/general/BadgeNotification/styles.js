export default (theme) => ({
    root: {
        display: 'inline-block',
        textAlign: 'center',
        backgroundColor: theme.colors.orange500,
        position: 'relative',
    },
    Text: {
        padding: '0px 8px',
        verticalAlign: 'middle',
        color: theme.colors.neutralBase,
    },
    medium: {
        height: 24,
        minWidth: 24,
        borderRadius: 24,
        '& $Text': {
            lineHeight: '24px',
        },
    },
    small: {
        height: 18,
        minWidth: 18,
        borderRadius: 18,
        '& $Text': {
            lineHeight: '18px',
        },
    },
});
