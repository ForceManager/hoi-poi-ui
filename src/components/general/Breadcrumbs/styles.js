export default (theme) => ({
    root: {
        display: 'flex',
        alignItems: 'center',
    },
    item: {
        marginRight: 8,
        '& span': {
            color: theme.colors.textLight.secondary,
        },
        '&:last-child': {
            margin: 0,
            '& span': {
                color: theme.colors.textLight.primary,
            },
        },
    },
    clickables: {
        '& $item': {
            cursor: 'pointer',
            '&:hover': {
                '& span': {
                    color: theme.colors.blue500,
                },
                '&:last-child': {
                    '& span': {
                        color: theme.colors.textLight.primary,
                    },
                },
            },
            '&:last-child': {
                cursor: 'inherit',
            },
        },
        '& $divider': {
            cursor: 'inherit',
        },
    },
    divider: {},
});
