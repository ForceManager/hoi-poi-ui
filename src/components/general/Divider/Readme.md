Default:

```jsx
import { Divider } from 'hoi-poi-ui';

<div>
    <Divider />
</div>;
```

### Component tree

---

-   root - root element
