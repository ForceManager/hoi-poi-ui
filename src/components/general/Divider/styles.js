export default (theme) => ({
    root: {
        display: 'block',
        height: 0,
        borderTop: `1px solid ${theme.colors.neutral400}`,
    },
});
