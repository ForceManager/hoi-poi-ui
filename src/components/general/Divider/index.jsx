import React, { memo, useMemo } from 'react';
import classNames from 'classnames';
import { getOverrides, useClasses } from '../../../utils/overrides';

import { createUseStyles } from '../../../utils/styles';
import styles from './styles';

const useStyles = createUseStyles(styles, 'Divider');

const Divider = ({
    classes: classesProp,
    overrides: overridesProp,
    className: classNameProp,
    ...props
}) => {
    const classes = useClasses(useStyles, classesProp);
    const override = getOverrides(overridesProp, Divider.overrides);

    const rootProps = useMemo(
        () => ({
            ...props,
            className: classNames(classes.root, classNameProp),
        }),
        [classNameProp, classes.root, props],
    );

    return <div {...rootProps} {...override.root} />;
};

export default memo(Divider);
