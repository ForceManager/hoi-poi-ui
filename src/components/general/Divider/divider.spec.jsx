import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HoiPoiProvider } from '../../../utils/styles';
import Divider from '../Divider';

describe('Divider', () => {
    test('default match', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <Divider />
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
