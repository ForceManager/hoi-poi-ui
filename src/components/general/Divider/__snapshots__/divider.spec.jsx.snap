// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider default match 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "aqua100": "#e4fafd",
          "aqua200": "#cbf3fc",
          "aqua300": "#AEE5F8",
          "aqua400": "#98d5f1",
          "aqua500": "#76bde8",
          "aqua600": "#5695c7",
          "aqua700": "#3b71a7",
          "aqua900": "#16386f",
          "black": "#3c4248",
          "blue100": "#d3e3fe",
          "blue200": "#a7c5fe",
          "blue300": "#7BA4FE",
          "blue400": "#5a89fd",
          "blue500": "#245dfd",
          "blue600": "#1a47d9",
          "blue700": "#1234b6",
          "blue900": "#061879",
          "green100": "#f4fad2",
          "green200": "#e7f6a6",
          "green300": "#A7E675",
          "green400": "#aecd50",
          "green500": "#86ad20",
          "green600": "#6e9417",
          "green700": "#237c10",
          "green900": "#345306",
          "neutral100": "#f9f9fa",
          "neutral150": "#f6f7f9",
          "neutral200": "#f4f5f6",
          "neutral300": "#e9ecee",
          "neutral400": "#dfe2e5",
          "neutral500": "#d4d8dc",
          "neutral600": "#a9b1b9",
          "neutral700": "#7d8a96",
          "neutral800": "#526373",
          "neutral900": "#273c50",
          "neutralBase": "#ffffff",
          "orange100": "#fff1cc",
          "orange200": "#ffde99",
          "orange300": "#FFC766",
          "orange400": "#ffb13f",
          "orange500": "#ff8c00",
          "orange600": "#db6e00",
          "orange700": "#b75400",
          "orange900": "#7a2c00",
          "overlay": "rgba(33,33,33,0.55)",
          "purple100": "#fae3ed",
          "purple200": "#f6c8e1",
          "purple300": "#E5A5CE",
          "purple400": "#cc84b8",
          "purple500": "#aa5b9c",
          "purple600": "#92428b",
          "purple700": "#792d7a",
          "purple900": "#441151",
          "red100": "#ffe5d5",
          "red200": "#ffc5ac",
          "red300": "#FF9E82",
          "red400": "#ff7863",
          "red500": "#ff3b30",
          "red600": "#db2328",
          "red700": "#b7182a",
          "red900": "#7a0929",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "turquoise100": "#c9fbf1",
          "turquoise200": "#95f8ec",
          "turquoise300": "#5EEBE5",
          "turquoise400": "#36d2d8",
          "turquoise500": "#00a8be",
          "turquoise600": "#0083a3",
          "turquoise700": "#006388",
          "turquoise900": "#00335b",
          "yellow100": "#fdf6cb",
          "yellow200": "#fcec97",
          "yellow300": "#F8DC63",
          "yellow400": "#f1cb3c",
          "yellow500": "#e9b200",
          "yellow600": "#c89400",
          "yellow700": "#a77700",
          "yellow900": "#6f4a00",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "boxShadow24": Object {
            "boxShadow": "0px 18px 40px rgba(39, 60, 80, 0.4)",
          },
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "badges": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 10,
            "fontWeight": 400,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "body": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 400,
            "letterSpacing": 0.6,
            "lineHeight": "20px",
          },
          "body1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": 400,
            "letterSpacing": 0.5,
            "lineHeight": "28px",
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0.6,
            "lineHeight": "16px",
          },
          "caption": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 400,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "captionMedium": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 500,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 14,
          "fontSizeSmall": 11,
          "h1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 96,
            "fontWeight": 300,
            "letterSpacing": -1.5,
            "lineHeight": "112px",
          },
          "h2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 60,
            "fontWeight": 300,
            "letterSpacing": -0.5,
            "lineHeight": "71px",
          },
          "h3": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 48,
            "fontWeight": 400,
            "lineHeight": "57px",
          },
          "h4": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 34,
            "fontWeight": 400,
            "letterSpacing": 0.25,
            "lineHeight": "40px",
          },
          "h5": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 24,
            "fontWeight": 400,
            "lineHeight": "28px",
          },
          "h6": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "fontWeight": 500,
            "letterSpacing": 0.25,
            "lineHeight": "24px",
          },
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "overline": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 500,
            "letterSpacing": 2,
            "lineHeight": "16px",
            "textTransform": "uppercase",
          },
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
          "subtitle": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0.09,
            "lineHeight": "20px",
          },
          "subtitle1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": 400,
            "letterSpacing": 0.15,
            "lineHeight": "24px",
          },
        },
        "utils": Object {
          "scrollbar": Object {
            "&::-webkit-scrollbar": Object {
              "height": 17,
              "width": 17,
            },
            "&::-webkit-scrollbar-thumb": Object {
              "background": "#dfe2e5",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
            "&::-webkit-scrollbar-thumb:window-inactive": Object {
              "background": "#dfe2e5",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
          },
          "selection": Object {
            "& ::-moz-selection": Object {
              "backgroundColor": "#ffde99",
            },
            "& ::selection": Object {
              "backgroundColor": "#ffde99",
            },
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.9.2",
        }
      }
    >
      <Divider>
        <div
          className="HoiPoi__Divider__root-1-0"
        />
      </Divider>
    </JssProvider>
  </ThemeProvider>
</Component>
`;
